source/OKR-Tests/ObjectiveRESTfulControllerTest.class.st

```diff
@@ -0,0 +1,231 @@
+Class {
+	#name : #ObjectiveRESTfulControllerTest,
+	#superclass : #ResourceRESTfulControllerTest,
+	#instVars : [
+		'objectiveManagementSystem'
+	],
+	#category : #'OKR-Tests-Controllers'
+}
+
+{ #category : #'private - support' }
+ObjectiveRESTfulControllerTest >> baseUrl [
+
+	^ 'https://api.mercap.net' asZnUrl
+]
+
+{ #category : #'private - support' }
+ObjectiveRESTfulControllerTest >> createObjectiveBasedOn: anObjective [
+
+	^ resourceController
+		createObjectiveBasedOn: ( self requestToPOST: anObjective as: self defaultObjectiveMediaType )
+		within: self newHttpRequestContext
+]
+
+{ #category : #'private - support' }
+ObjectiveRESTfulControllerTest >> defaultObjectiveMediaType [
+
+	^ resourceController specification objectiveVersion1dot0dot0MediaType
+]
+
+{ #category : #'private - accessing' }
+ObjectiveRESTfulControllerTest >> emptyObjectiveJson [
+
+	^ NeoJSONObject new
+		at: 'description' put: 'First one'; 
+		at: 'keyResults' put: Array new; 
+		asString
+]
+
+{ #category : #'private - support' }
+ObjectiveRESTfulControllerTest >> getAllObjectives [
+
+	^ resourceController
+		getObjectivesBasedOn: ( self requestToGet: self resourceUrl accepting: '*/*' )
+		within: self newHttpRequestContext
+]
+
+{ #category : #'private - support' }
+ObjectiveRESTfulControllerTest >> getObjectiveIdentifiedAs: anIdentifier [
+
+	^ resourceController
+		getObjectiveBasedOn: ( self requestToGetResourceIdentifiedBy: anIdentifier accepting: self defaultObjectiveMediaType )
+		within: self newHttpRequestContext
+]
+
+{ #category : #'private - support' }
+ObjectiveRESTfulControllerTest >> registerObjective: anObjective [
+
+	objectiveManagementSystem startManagingObjective: anObjective
+]
+
+{ #category : #'private - support' }
+ObjectiveRESTfulControllerTest >> requestToDeleteObjective: anObjectiveLocation [
+
+	^ TeaRequest fromZnRequest: ( ZnRequest delete: anObjectiveLocation )
+]
+
+{ #category : #'private - support' }
+ObjectiveRESTfulControllerTest >> requestToGet: aUrl accepting: anAcceptHeader [
+
+	^ TeaRequest
+		fromZnRequest:
+			( ( ZnRequest get: aUrl )
+				setAccept: anAcceptHeader;
+				yourself )
+]
+
+{ #category : #'private - support' }
+ObjectiveRESTfulControllerTest >> requestToUpdateObjective: aPeriodLocation with: anUpdatePeriod [
```

Comment:
Rename aPeriodLocation and anUpdatePeriod to reference Objective instead of Period